File: Csharp/Lab8/Lab8/Lab8/cng/Model/Collections/Comparators/PriceComparatorToGreater.cs
using System;
using System.Collections;
using System.Collections.Generic;
using Lab8.cng.Model.Entity.Abstracts;

namespace Lab8.cng.Model.Collections.Comparators
{
    public class PriceComparatorToGreater : IComparer
    {
        
        public int Compare(object x, object y)
        {
            Toy toy1 = (Toy) x;
            Toy toy2 = (Toy) y;
            return (int) (toy1.GetPrice() - toy2.GetPrice());
        }
    }
}

File: Kotlin/Lab8/src/by/bntu/fitr/povt/cng/model/collections/comparators/PriceComparatorToGreater.java
package by.bntu.fitr.povt.cng.model.collections.comparators;

import by.bntu.fitr.povt.cng.model.entity.abstracts.Toy;

import java.util.Comparator;

public class PriceComparatorToGreater implements Comparator<Toy> {


    @Override
    public int compare(Toy o1, Toy o2) {
        return (int) (o1.getPrice() - o2.getPrice());
    }
}
